### PR TITLE
[mdns] Cancel removal task on manual scan

### PR DESCRIPTION
**Pull request by cweitkamp**
- Cancel removal task on manual scan

Signed-off-by: Christoph Weitkamp <github@christophweitkamp.de>

### CODE REVIEW

File: bundles/org.openhab.core.config.discovery.mdns/src/main/java/org/openhab/core/config/discovery/mdns/internal/MDNSDiscoveryService.java [hunk 1]
```diff
@@ -230,19 +211,40 @@ private void considerService(ServiceEvent serviceEvent) {
...
-                    } catch (Exception e) {
-                        logger.error("Participant '{}' threw an exception", participant.getClass().getName(), e);
-                    }
+                    createDicoveryResult(participant, serviceEvent.getInfo());
+                }
+            }
+        }
+    }
+
+    private void createDicoveryResult(MDNSDiscoveryParticipant participant, ServiceInfo serviceInfo) {
```
Comment (jlaur) [inline bundles/org.openhab.core.config.discovery.mdns/src/main/java/org/openhab/core/config/discovery/mdns/internal/MDNSDiscoveryService.java:220]: Typo: createDi**s**coveryResult

File: bundles/org.openhab.core.config.discovery.mdns/src/main/java/org/openhab/core/config/discovery/mdns/internal/MDNSDiscoveryService.java [hunk 2]
```diff
@@ -230,19 +211,40 @@ private void considerService(ServiceEvent serviceEvent) {
...
+                final DiscoveryResult resultNew = getLocalizedDiscoveryResult(result,
+                        FrameworkUtil.getBundle(participant.getClass()));
+                thingDiscovered(resultNew);
+            }
+        } catch (Exception e) {
+            logger.error("Participant '{}' threw an exception", participant.getClass().getName(), e);
+        }
+    }
+
+    private void removeDicoveryResult(MDNSDiscoveryParticipant participant, ServiceInfo serviceInfo) {
```
Comment (jlaur) [inline bundles/org.openhab.core.config.discovery.mdns/src/main/java/org/openhab/core/config/discovery/mdns/internal/MDNSDiscoveryService.java:234]: Typo: removeDi**s**coveryResult

File: bundles/org.openhab.core.config.discovery.mdns/src/main/java/org/openhab/core/config/discovery/mdns/internal/MDNSDiscoveryService.java (hunk 1 above)
Comment (cweitkamp) [reply bundles/org.openhab.core.config.discovery.mdns/src/main/java/org/openhab/core/config/discovery/mdns/internal/MDNSDiscoveryService.java]: Nice catch. Fixed.

**Comment by cweitkamp** [comment]
I added a unit test to cover JmDNS service events.

**Comment by andrewfg** [comment]
LGTM :)

**Review by kaikreuzer — APPROVED** [review]
Thanks!

**Pull request merged**